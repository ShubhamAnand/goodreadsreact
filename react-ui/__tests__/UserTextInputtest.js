import React from 'react';
import LoaderCustom from '../src/components/Loader';
import { shallow,configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'


configure({ adapter: new Adapter() })

describe("Testing LoaderCustom Component", () => {
    it('should render correctly', () => {
        const wrapper = shallow(
            <LoaderCustom store={store} onChange={()=>{}} autoFocus="true" value="12"/>)
            expect(wrapper).toMatchSnapshot();
      });
    
      it('should display passed text correctly', () => {
        const textwrapper = shallow(
            <UserTextInput value="This is value"/>);
            expect(textwrapper).toMatchSnapshot();
            expect(textwrapper.find('TextInput').props().value).toBe('This is value');
      });

      //triggering events
      it('should update state on focus', () => {
        const textwrapperonchange = shallow(
            <UserTextInput value="This is value"/>)
             //console.log(textwrapperonchange.debug());
             expect(textwrapperonchange).toMatchSnapshot();
            (textwrapperonchange.at(0).find('TextInput').simulate('focus'));
            expect(textwrapperonchange).toMatchSnapshot();
           // expect(textwrapperonchange.state().inFocus).toBe(true);
      });

      it('should update state on blur', () => {
        const textwrapperonfocus = shallow(
            <UserTextInput value="This is value"/>);
            (textwrapperonfocus.find('TextInput').simulate('blur'));
            expect(textwrapperonfocus).toMatchSnapshot();
            expect(textwrapperonfocus.state().inFocus).toBe(false);
      });
});

