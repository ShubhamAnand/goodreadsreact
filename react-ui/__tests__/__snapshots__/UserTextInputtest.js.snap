// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing User Text Input Component should display passed text correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserTextInput
    value="This is value"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoFocus={undefined}
        numberOfLines={1}
        onBlur={[Function]}
        onChangeText={undefined}
        onFocus={[Function]}
        returnKeyType="next"
        selectionColor="rgba(0,0,0,0.5)"
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "height": 45,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
        value="This is value"
      />,
      "style": Array [
        Object {
          "borderBottomLeftRadius": 4,
          "borderBottomRightRadius": 4,
          "borderBottomWidth": 1,
          "borderColor": "#EEEEEE",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowFontScaling": true,
        "autoCapitalize": "none",
        "autoFocus": undefined,
        "numberOfLines": 1,
        "onBlur": [Function],
        "onChangeText": undefined,
        "onFocus": [Function],
        "returnKeyType": "next",
        "selectionColor": "rgba(0,0,0,0.5)",
        "style": Object {
          "color": "#000000",
          "fontSize": 16,
          "height": 45,
          "width": "100%",
        },
        "underlineColorAndroid": "transparent",
        "value": "This is value",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoFocus={undefined}
          numberOfLines={1}
          onBlur={[Function]}
          onChangeText={undefined}
          onFocus={[Function]}
          returnKeyType="next"
          selectionColor="rgba(0,0,0,0.5)"
          style={
            Object {
              "color": "#000000",
              "fontSize": 16,
              "height": 45,
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
          value="This is value"
        />,
        "style": Array [
          Object {
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderBottomWidth": 1,
            "borderColor": "#EEEEEE",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoCapitalize": "none",
          "autoFocus": undefined,
          "numberOfLines": 1,
          "onBlur": [Function],
          "onChangeText": undefined,
          "onFocus": [Function],
          "returnKeyType": "next",
          "selectionColor": "rgba(0,0,0,0.5)",
          "style": Object {
            "color": "#000000",
            "fontSize": 16,
            "height": 45,
            "width": "100%",
          },
          "underlineColorAndroid": "transparent",
          "value": "This is value",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing User Text Input Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserTextInput
    autoFocus="true"
    onChange={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
    value="12"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoFocus="true"
        numberOfLines={1}
        onBlur={[Function]}
        onChangeText={[Function]}
        onFocus={[Function]}
        returnKeyType="next"
        selectionColor="rgba(0,0,0,0.5)"
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "height": 45,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
        value="12"
      />,
      "style": Array [
        Object {
          "borderBottomLeftRadius": 4,
          "borderBottomRightRadius": 4,
          "borderBottomWidth": 1,
          "borderColor": "#EEEEEE",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowFontScaling": true,
        "autoCapitalize": "none",
        "autoFocus": "true",
        "numberOfLines": 1,
        "onBlur": [Function],
        "onChangeText": [Function],
        "onFocus": [Function],
        "returnKeyType": "next",
        "selectionColor": "rgba(0,0,0,0.5)",
        "style": Object {
          "color": "#000000",
          "fontSize": 16,
          "height": 45,
          "width": "100%",
        },
        "underlineColorAndroid": "transparent",
        "value": "12",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoFocus="true"
          numberOfLines={1}
          onBlur={[Function]}
          onChangeText={[Function]}
          onFocus={[Function]}
          returnKeyType="next"
          selectionColor="rgba(0,0,0,0.5)"
          style={
            Object {
              "color": "#000000",
              "fontSize": 16,
              "height": 45,
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
          value="12"
        />,
        "style": Array [
          Object {
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderBottomWidth": 1,
            "borderColor": "#EEEEEE",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoCapitalize": "none",
          "autoFocus": "true",
          "numberOfLines": 1,
          "onBlur": [Function],
          "onChangeText": [Function],
          "onFocus": [Function],
          "returnKeyType": "next",
          "selectionColor": "rgba(0,0,0,0.5)",
          "style": Object {
            "color": "#000000",
            "fontSize": 16,
            "height": 45,
            "width": "100%",
          },
          "underlineColorAndroid": "transparent",
          "value": "12",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing User Text Input Component should update state on blur 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserTextInput
    value="This is value"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoFocus={undefined}
        numberOfLines={1}
        onBlur={[Function]}
        onChangeText={undefined}
        onFocus={[Function]}
        returnKeyType="next"
        selectionColor="rgba(0,0,0,0.5)"
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "height": 45,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
        value="This is value"
      />,
      "style": Array [
        Object {
          "borderBottomLeftRadius": 4,
          "borderBottomRightRadius": 4,
          "borderBottomWidth": 1,
          "borderColor": "#EEEEEE",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowFontScaling": true,
        "autoCapitalize": "none",
        "autoFocus": undefined,
        "numberOfLines": 1,
        "onBlur": [Function],
        "onChangeText": undefined,
        "onFocus": [Function],
        "returnKeyType": "next",
        "selectionColor": "rgba(0,0,0,0.5)",
        "style": Object {
          "color": "#000000",
          "fontSize": 16,
          "height": 45,
          "width": "100%",
        },
        "underlineColorAndroid": "transparent",
        "value": "This is value",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoFocus={undefined}
          numberOfLines={1}
          onBlur={[Function]}
          onChangeText={undefined}
          onFocus={[Function]}
          returnKeyType="next"
          selectionColor="rgba(0,0,0,0.5)"
          style={
            Object {
              "color": "#000000",
              "fontSize": 16,
              "height": 45,
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
          value="This is value"
        />,
        "style": Array [
          Object {
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderBottomWidth": 1,
            "borderColor": "#EEEEEE",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoCapitalize": "none",
          "autoFocus": undefined,
          "numberOfLines": 1,
          "onBlur": [Function],
          "onChangeText": undefined,
          "onFocus": [Function],
          "returnKeyType": "next",
          "selectionColor": "rgba(0,0,0,0.5)",
          "style": Object {
            "color": "#000000",
            "fontSize": 16,
            "height": 45,
            "width": "100%",
          },
          "underlineColorAndroid": "transparent",
          "value": "This is value",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing User Text Input Component should update state on focus 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserTextInput
    value="This is value"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoFocus={undefined}
        numberOfLines={1}
        onBlur={[Function]}
        onChangeText={undefined}
        onFocus={[Function]}
        returnKeyType="next"
        selectionColor="rgba(0,0,0,0.5)"
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "height": 45,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
        value="This is value"
      />,
      "style": Array [
        Object {
          "borderBottomLeftRadius": 4,
          "borderBottomRightRadius": 4,
          "borderBottomWidth": 1,
          "borderColor": "#EEEEEE",
          "borderTopLeftRadius": 4,
          "borderTopRightRadius": 4,
          "borderWidth": 1,
          "flexDirection": "row",
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowFontScaling": true,
        "autoCapitalize": "none",
        "autoFocus": undefined,
        "numberOfLines": 1,
        "onBlur": [Function],
        "onChangeText": undefined,
        "onFocus": [Function],
        "returnKeyType": "next",
        "selectionColor": "rgba(0,0,0,0.5)",
        "style": Object {
          "color": "#000000",
          "fontSize": 16,
          "height": 45,
          "width": "100%",
        },
        "underlineColorAndroid": "transparent",
        "value": "This is value",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoFocus={undefined}
          numberOfLines={1}
          onBlur={[Function]}
          onChangeText={undefined}
          onFocus={[Function]}
          returnKeyType="next"
          selectionColor="rgba(0,0,0,0.5)"
          style={
            Object {
              "color": "#000000",
              "fontSize": 16,
              "height": 45,
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
          value="This is value"
        />,
        "style": Array [
          Object {
            "borderBottomLeftRadius": 4,
            "borderBottomRightRadius": 4,
            "borderBottomWidth": 1,
            "borderColor": "#EEEEEE",
            "borderTopLeftRadius": 4,
            "borderTopRightRadius": 4,
            "borderWidth": 1,
            "flexDirection": "row",
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoCapitalize": "none",
          "autoFocus": undefined,
          "numberOfLines": 1,
          "onBlur": [Function],
          "onChangeText": undefined,
          "onFocus": [Function],
          "returnKeyType": "next",
          "selectionColor": "rgba(0,0,0,0.5)",
          "style": Object {
            "color": "#000000",
            "fontSize": 16,
            "height": 45,
            "width": "100%",
          },
          "underlineColorAndroid": "transparent",
          "value": "This is value",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing User Text Input Component should update state on focus 2`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserTextInput
    value="This is value"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <TextInput
        allowFontScaling={true}
        autoCapitalize="none"
        autoFocus={undefined}
        numberOfLines={1}
        onBlur={[Function]}
        onChangeText={undefined}
        onFocus={[Function]}
        returnKeyType="next"
        selectionColor="rgba(0,0,0,0.5)"
        style={
          Object {
            "color": "#000000",
            "fontSize": 16,
            "height": 45,
            "width": "100%",
          }
        }
        underlineColorAndroid="transparent"
        value="This is value"
      />,
      "style": Array [
        Object {
          "borderColor": "#2196F3",
          "borderRadius": 4,
          "borderWidth": 1,
          "elevation": 1,
          "flexDirection": "row",
        },
        undefined,
      ],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "allowFontScaling": true,
        "autoCapitalize": "none",
        "autoFocus": undefined,
        "numberOfLines": 1,
        "onBlur": [Function],
        "onChangeText": undefined,
        "onFocus": [Function],
        "returnKeyType": "next",
        "selectionColor": "rgba(0,0,0,0.5)",
        "style": Object {
          "color": "#000000",
          "fontSize": 16,
          "height": 45,
          "width": "100%",
        },
        "underlineColorAndroid": "transparent",
        "value": "This is value",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          autoFocus={undefined}
          numberOfLines={1}
          onBlur={[Function]}
          onChangeText={undefined}
          onFocus={[Function]}
          returnKeyType="next"
          selectionColor="rgba(0,0,0,0.5)"
          style={
            Object {
              "color": "#000000",
              "fontSize": 16,
              "height": 45,
              "width": "100%",
            }
          }
          underlineColorAndroid="transparent"
          value="This is value"
        />,
        "style": Array [
          Object {
            "borderColor": "#2196F3",
            "borderRadius": 4,
            "borderWidth": 1,
            "elevation": 1,
            "flexDirection": "row",
          },
          undefined,
        ],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "allowFontScaling": true,
          "autoCapitalize": "none",
          "autoFocus": undefined,
          "numberOfLines": 1,
          "onBlur": [Function],
          "onChangeText": undefined,
          "onFocus": [Function],
          "returnKeyType": "next",
          "selectionColor": "rgba(0,0,0,0.5)",
          "style": Object {
            "color": "#000000",
            "fontSize": 16,
            "height": 45,
            "width": "100%",
          },
          "underlineColorAndroid": "transparent",
          "value": "This is value",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
